ctms/app.py

```diff
@@ -690,6 +691,41 @@ def partial_update_ctms_contact(
     return get_ctms_response_or_404(db=db, email_id=email_id)
 
 
+@app.delete(
+    "/ctms/{primary_email}",
+    summary="Delete all contact information from primary email",
+    response_model=List[IdentityResponse],
+    responses={
+        404: {"model": NotFoundResponse},
+    },
+    tags=["Public"],
+)
+def delete_contact_by_primary_email(
+    primary_email: str,
+    db: Session = Depends(get_db),
+    api_client: ApiClientSchema = Depends(get_enabled_api_client),
+):
+    ids = all_ids(primary_email=primary_email.lower())
+    contacts = get_contacts_by_ids(db, **ids)
+
+    if not contacts:
+        raise HTTPException(status_code=404, detail=f"email {primary_email} not found!")
+
+    dropped = []
+
+    for contact in contacts:
+        delete_contact(db=db, email_id=contact.email.email_id)
+
+        dropped.append(
+            {
+                "primary_email": contact.email.primary_email,
+                "email_id": str(contact.email.email_id),
+            }
+        )
```

Comment:
Here `dropped` is not used anymore
You could log these infos though

Reply:
right, getting rid of it :+1: